Fix some instances of null array key access

mukeshpanchal27 opened the pull request:
<!--
Hi there! Thanks for contributing to WordPress!

Pull Requests in this GitHub repository **must** be linked to a ticket in the WordPress Core Trac instance (https://core.trac.wordpress.org), and are only used for code review. **No pull requests will be merged on GitHub.**

See the WordPress Handbook page on using PRs for Code Review more information: https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/

If this is your first time contributing, you may also find reviewing these guides first to be helpful:
- FAQs for New Contributors: https://make.wordpress.org/core/handbook/tutorials/faq-for-new-contributors/
- Contributing with Code Guide: https://make.wordpress.org/core/handbook/contribute/
- WordPress Coding Standards: https://make.wordpress.org/core/handbook/best-practices/coding-standards/
- Inline Documentation Standards: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/
- Browser Support Policies: https://make.wordpress.org/core/handbook/best-practices/browser-support/
- Proper spelling and grammar related best practices: https://make.wordpress.org/core/handbook/best-practices/spelling/
-->

<!-- Insert a description of your changes here -->

Trac ticket: https://core.trac.wordpress.org/ticket/63957

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

mukeshpanchal27 committed:
Update is_registered method to handle null template name

mukeshpanchal27 committed:
Update class-wp-block-bindings-registry.php

mukeshpanchal27 committed:
Update class-wp-block-pattern-categories-registry.php

mukeshpanchal27 committed:
Change pattern_name parameter to nullable string

Update parameter type hint for is_registered method.

mukeshpanchal27 committed:
Update parameter types in is_registered method

mukeshpanchal27 committed:
Fix parameter type hint in is_registered method

mukeshpanchal27 commented:
Follow-up to #9849

swissspidy commented:
How did you encounter these? #9845 doesn't report them, which would indicate a lack of test coverage.

mukeshpanchal27 commented:
> How did you encounter these? #9845 doesn't report them, which would indicate a lack of test coverage.

Yes.

swissspidy commented:
Yes there is a lack of test coverage?

In that case, some tests would be great :-)

mukeshpanchal27 commented on src/wp-includes/class-wp-block-styles-registry.php:
```diff
@@ -181,12 +181,12 @@ public function get_registered_styles_for_block( $block_name ) {
 	 */
 	public function is_registered( $block_name, $block_style_name ) {
-		return isset( $this->registered_block_styles[ $block_name ][ $block_style_name ] );
+		return isset( $block_name, $block_style_name, $this->registered_block_styles[ $block_name ][ $block_style_name ] );
```
```suggestion
		return isset( $block_name, $this->registered_block_styles[ $block_name ][ $block_style_name ] );
```

mukeshpanchal27 committed:
Apply suggestions from code review

mukeshpanchal27 committed:
Add test for empty source name registration

Add test for is_registered method with empty source name

mukeshpanchal27 committed:
Add unit tests for WP_Block_Pattern_Categories_Registry::is_registered()

mukeshpanchal27 committed:
Add tests for WP_Block_Patterns_Registry methods

mukeshpanchal27 committed:
Add unit tests for WP_Block_Styles_Registry::is_registered

This file contains unit tests for the WP_Block_Styles_Registry class, specifically testing the is_registered() method with various scenarios.

mukeshpanchal27 committed:
Update WpBlockTemplatesRegistry.php

mukeshpanchal27 commented on tests/phpunit/tests/block-pattern/wpBlockPatternCategoriesRegistry.php:
```diff
@@ -0,0 +1,33 @@
+	public function test_is_registered_with_empty_param() {
+		$registry = WP_Block_Pattern_Categories_Registry::get_instance();
+
+		// Should return false when called with empty string.
+		$this->assertFalse( $registry->is_registered( '' ) );
+
+		// Should return false when called with null.
+		$this->assertFalse( $registry->is_registered( null ) );
+
+		// Should return false when called with false.
+		$this->assertFalse( $registry->is_registered( false ) );
+
+		// Should return false when called with 0.
+		$this->assertFalse( $registry->is_registered( 0 ) );
```
```suggestion
		$this->assertFalse( $registry->is_registered( '' ) );
		$this->assertFalse( $registry->is_registered( null ) );
```

mukeshpanchal27 commented on tests/phpunit/tests/block-pattern/wpBlockPatternsRegistry.php:
```diff
@@ -0,0 +1,81 @@
+	public function test_is_registered_with_invalid_pattern_name() {
+		$this->assertFalse( $this->registry->is_registered( '' ) );
+		$this->assertFalse( $this->registry->is_registered( null ) );
+		$this->assertFalse( $this->registry->is_registered( false ) );
+		$this->assertFalse( $this->registry->is_registered( 0 ) );
```
```suggestion
```

mukeshpanchal27 committed:
Remove bool check

mukeshpanchal27 committed:
Run unit tests against PHP 8.5

swissspidy commented on src/wp-includes/class-wp-block-styles-registry.php:
```diff
@@ -181,12 +181,12 @@ public function get_registered_styles_for_block( $block_name ) {
-	 * @param string $block_style_name Block style name.
+	 * @param string|null $block_name       Block type name including namespace.
+	 * @param string|null $block_style_name Block style name.
 	 * @return bool True if the block style is registered, false otherwise.
 	 */
 	public function is_registered( $block_name, $block_style_name ) {
-		return isset( $this->registered_block_styles[ $block_name ][ $block_style_name ] );
+		return isset( $block_name, $this->registered_block_styles[ $block_name ][ $block_style_name ] );
```
If block_style_name really can be null you need to check it here too

swissspidy commented on tests/phpunit/tests/block-bindings/wpBlockBindingsRegistry.php:
```diff
@@ -344,4 +344,16 @@ public function test_is_registered_for_known_source() {
 		$this->assertTrue( $result );
 	}
+
+	/**
+	 * Should return false when checking registration with an empty source name.
+	 *
+	 * @ticket 63957
+	 *
+	 * @covers WP_Block_Bindings_Registry::is_registered
+	 */
+	public function test_is_registered_with_empty_source_name() {
+		$result = $this->registry->is_registered( '' );
+		$this->assertFalse( $result );
+	}
```
But the whole PR is about `null` checks, not `false`? 🤔

mukeshpanchal27 replied:
Did i miss anything? The function return bool value

swissspidy replied:
You're testing `is_registered( '' )`, not `is_registered( null )`

swissspidy commented on tests/phpunit/tests/block-pattern/wpBlockPatternsRegistry.php:
There is already `tests/phpunit/tests/blocks/wpBlockPatternsRegistry.php`, let's consolidate tests there

swissspidy commented on tests/phpunit/tests/block-styles/wpBlockStylesRegistry.php:
Ditto here, there is already `tests/phpunit/tests/blocks/wpBlockStylesRegistry.php`

swissspidy commented:
> Yes there is a lack of test coverage?
> 
> In that case, some tests would be great :-)

Curious to see any stack traces where these happen in action too, to find the root cause

mukeshpanchal27 committed:
Check for null param

mukeshpanchal27 committed:
Move tests

mukeshpanchal27 committed:
Remove empty check
